tests/MongoDB.Driver.Examples/InsertDataWithEncryptedFieldsWithAutoEncryption.cs

```diff
@@ -0,0 +1,292 @@
+                        { "policyNumber", 123142 },
+                        { "provider", "MaestCare" }
+                    }
+                }
+            };
+
+        // https://gist.github.com/DmitryLukyanov/8417d1f5c30f12ffeb2be29056aff455#file-insertdatawithencryptedfields-cs
+        [Fact]
+        public void InsertDataWithEncryptedFields()
+        {
+            var keyVaultNamespace = CollectionNamespace.FromFullName("encryption.__keyVault");
+            var medicalRecordsNamespace = CollectionNamespace.FromFullName("medicalRecords.patients");
+
+            var keyIdBase64 = ""; // paste the generated key in Standard GuidRepresentation form here
+
+            // Construct a JSON Schema
+            var schema = JsonSchemaCreator.CreateJsonSchema(keyIdBase64);
+
+            // Construct an auto-encrypting client
+            var autoEncryptingClient = CreateAutoEncryptingClient(
+                medicalRecordsNamespace,
+                keyVaultNamespace,
+                schema);
+            var collection = autoEncryptingClient
+                .GetDatabase(medicalRecordsNamespace.DatabaseNamespace.DatabaseName)
+                .GetCollection<BsonDocument>(medicalRecordsNamespace.CollectionName);
+
+            var ssnQuery = Builders<BsonDocument>.Filter.Eq("ssn", __sampleSsnValue);
+
+            // Insert a document into the collection
+            collection.UpdateOne(ssnQuery, new BsonDocument("$set", __sampleDocFields), new UpdateOptions() { IsUpsert = true });
+            Console.WriteLine("Successfully upserted the sample document!");
+
+            // Query by SSN field with auto-encrypting client
+            var result = collection.Find(ssnQuery).Single();
+
+            Console.WriteLine("Query by SSN (deterministically-encrypted field) using auto-encrypting client returned:\n" + result.ToJson());
+
+            // Query by SSN field with a normal non-auto-encrypting client
+            var nonAutoEncryptingClient = new MongoClient(__connectionString);
+            collection = nonAutoEncryptingClient
+              .GetDatabase(medicalRecordsNamespace.DatabaseNamespace.DatabaseName)
+              .GetCollection<BsonDocument>(medicalRecordsNamespace.CollectionName);
+            result = collection.Find(ssnQuery).FirstOrDefault();
+            if (result != null)
+            {
+                throw new Exception("Expected no document to be found but one was found.");
+            }
+
+            // Query by name field with a normal non-auto-encrypting client
+            var nameQuery = Builders<BsonDocument>.Filter.Eq("name", __sampleNameValue);
+            result = collection.Find(nameQuery).FirstOrDefault();
+            if (result == null)
+            {
+                throw new Exception("Expected the document to be found but none was found.");
+            }
+
+            Console.WriteLine($"Query by name (non-encrypted field) using non-auto-encrypting client returned:\n" + result.ToJson());
```

Comment:
This line used to be:

    Console.WriteLine($"Query by name (non-encrypted field) using non-auto-encrypting client returned:\n {result}.");

which seems better than the latest version.

Note that `BsonDocument.ToString` calls `ToJson` so the call to `ToJson` can be implied. Even if you want to be explicit you can also write:

    Console.WriteLine($"Query by name (non-encrypted field) using non-auto-encrypting client returned:\n {result.ToJson()}.");